src/irmin-pack/dict.ml

```diff
@@ -98,4 +98,4 @@ let v_no_cache ~fresh ~shared:_ ~readonly file =
   refill ~from:0L t;
   t
 
-let v = with_cache ~clear ~v:v_no_cache "store.dict"
+let v = with_cache ~clear ~v:(fun ~aux:() -> v_no_cache) "store.dict" ~aux:()
```

Comment:
to make sure this works as intended, I'm in favour of adding an explicit staging operations, as in https://github.com/mirage/index/commit/fb182ab2c1ae8c3048e1f2661d063d7e7b1bce09

Reply:
Agreed. This is now done.